app/javascript/hooks/forms/authentication/useSignUpForm.js

```diff
@@ -0,0 +1,86 @@
+import * as yup from 'yup';
+import { yupResolver } from '@hookform/resolvers/yup';
+import { useTranslation } from 'react-i18next';
+import { useForm } from 'react-hook-form';
+
+export default function useSignUpForm(_config = {}) {
+  const { t } = useTranslation();
+
+  const fields = {
+    name: {
+      label: t('forms.user.signup.fields.full_name.label'),
+      placeHolder: t('forms.user.signup.fields.full_name.placeholder'),
+      controlId: 'signupFormFullName',
+      hookForm: {
+        id: 'name',
+      },
+    },
+    email: {
+      label: t('forms.user.signup.fields.email.label'),
+      placeHolder: t('forms.user.signup.fields.email.placeholder'),
+      controlId: 'signupFormEmail',
+      hookForm: {
+        id: 'email',
+      },
+    },
+    password: {
+      label: t('forms.user.signup.fields.password.label'),
+      placeHolder: t('forms.user.signup.fields.password.placeholder'),
+      controlId: 'signupFormPwd',
+      hookForm: {
+        id: 'password',
+        validations: {
+          deps: ['password_confirmation'],
+        },
+      },
+    },
+    password_confirmation: {
+      label: t('forms.user.signup.fields.password_confirmation.label'),
+      placeHolder: t('forms.user.signup.fields.password_confirmation.placeholder'),
+      controlId: 'signupFormPwdConfirm',
+      hookForm: {
+        id: 'password_confirmation',
+        validations: {
+          deps: ['password'],
+        },
+      },
+    },
+  };
+
+  const validationSchema = yup.object({
+    name: yup.string().required('forms.user.signup.validations.full_name.required')
+      .min(2, 'forms.user.signup.validations.full_name.min')
```

Comment:
We can localize validations from here now,
I just left as is for compatibility,
You can carry any WIP on localizing validations and we can make the change later on,